packages/zendesk-adapter/src/userUtils.ts

```diff
@@ -0,0 +1,75 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import Joi from 'joi'
+import { logger } from '@salto-io/logging'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import { collections } from '@salto-io/lowerdash'
+
+const log = logger(module)
+const { toArrayAsync } = collections.asynciterable
+const { makeArray } = collections.array
+
+type User = {
+  id: number
+  email: string
+  role: string
+  // eslint-disable-next-line camelcase
+  custom_role_id: number
+}
+
+const EXPECTED_USER_SCHEMA = Joi.array().items(Joi.object({
+  id: Joi.number().required(),
+  email: Joi.string().required(),
+  role: Joi.string(),
+  custom_role_id: Joi.number(),
+}).unknown(true)).required()
+
+const areUsers = (values: unknown): values is User[] => {
+  const { error } = EXPECTED_USER_SCHEMA.validate(values)
+  if (error !== undefined) {
+    log.warn(`Received an invalid response for the users values: ${error.message}`)
+    return false
+  }
+  return true
+}
+
+export const getUsersFunc = (paginator: clientUtils.Paginator):() => Promise<User[]> => {
+  let calculatedUsers: User[]
+
+  const getUsers = async (): Promise<User[]> => {
```

Comment:
this works, but I think it'd be cleaner to abstract away the fact that there is a wrapping function and an inner function here - so we'd want to have the inner function get the paginator and not the outer one, something like:
```suggestion
const getUsersFunc = (): (paginator: clientUtils.Paginator) => Promise<User[]> => {
  let calculatedUsers: User[]

  const getUsers = async (paginator: clientUtils.Paginator): Promise<User[]> => {
```
and after it was implemented
```
export const getUsers = getUsersFunc()
```

then in the callers you'd just call 
```
const users = await getUsers(paginator)
```
as you do today

Reply:
Cool, thanks :)